FHEM/00_SIGNALduino.pm

```diff
@@ -3827,8 +3848,27 @@ sub SetSens {
 }
 
 
+############################################################################
+########## Section & cc1101 settings for manufacturer specific´s  ##########
+package cc1101_manspecific;
+
+sub Register {
+	my ($aVal) = @_;
 
-################################################################################################																																							
+	my %modes = (
+		# adress to write from cc1101 register
+		'address'        => ['00','02','03','04','05','06','07','08','0D','0E','0F','10','11','12','13','14','15','17','18','19','1B','1C','1D','23','25','26','2B'],
+		# manufacturer specific´s register values
+		'Kopp_FC'        => ['01','06','04','AA','54','0F','E0','00','21','65','6A','97','83','16','63','F8','47','0C','29','36','43','40','91','E9','00','11','3E'],
+		'Lacrosse_mode1' => ['01','46','02','2D','D4','FF','00','02','21','65','6A','89','5C','06','22','F8','56','00','18','16','43','68','91','EC','17','11','3E'],
+		'Lacrosse_mode2' => ['01','46','02','2D','D4','FF','00','02','21','65','6A','88','82','06','22','F8','56','00','18','16','43','68','91','EC','16','11','3E'],
+		'Lacrosse_mode4' => ['01','46','02','2D','D4','FF','00','02','21','65','6A','8C','65','06','22','F8','56','00','18','16','43','68','91','EC','16','11','3E'],
+		'PCA301'         => ['01','46','07','2D','D4','FF','00','02','21','6B','D0','88','0B','06','22','F8','53','00','18','16','43','68','91','ED','17','11','3E']
+	);
+	return ($modes{$aVal}, $modes{'address'});
```

Comment:
Wenn ich das richtig interpretiere, dann werden hier die Werte beider arrays ohne Unterscheidungsmöglichkeit in Form von Duplikaten zurück gegeben oder sehe ich das falsch?

Reply:
Hier geht zurück das Array mit den Werten von dem auszuwählenden Modus und die Adressen worin diese geschrieben werden müssen.

Reply:
Ah nee, ich verstehe, es wird eine ref auf den array zurück gegeben :) Hab ich nicht sofort geblickt :)

Reply:
Ja kann man so machen, vermutlich hätte ich sowas gemacht wie:

```
my %hash;
@hash{@{$modes{'address'}}} = @{$modes{$aVal}};
```
Naja ist besstimt noch ein syntaxfehler. Dann hätte man einen hash gehabt in der Form

`$hash{00} => '01'`

Geht aber auch so. Das Problem mit der Rückgabe von zwei arrays kommentiere ich an der Stelle, an der ein Problem entstehen kann

Reply:
Diese Sub ist hinfällig weil diese entfernt wurde :-D